narrative_llm_agent/agents/analyst.py

```diff
@@ -6,62 +6,138 @@
 from langchain.vectorstores import Chroma
 from langchain.memory import ConversationBufferMemory, ReadOnlySharedMemory
 from langchain.chains import RetrievalQA
-from langchain.tools import BaseTool, tool
+from langchain.tools import tool
+import os
+from pathlib import Path
 
 
 class AnalystInput(BaseModel):
-    input: str = Field(description="query to look up KBase documentation, catalog or tutorials")
+    input: str = Field(
+        description="query to look up KBase documentation, catalog or tutorials"
+    )
+
+DEFAULT_CATALOG_DB_DIR: Path = Path(__file__).parent / "vector_db_app_catalog"
+DEFAULT_DOCS_DB_DIR: Path = Path(__file__).parent / "vector_db_kbase_docs"
 
 class AnalystAgent(KBaseAgent):
-    role="Computational Biologist and Geneticist"
-    goal="Analyze and interpret datasets, and make suggestions into next analysis steps."
-    backstory="""You are an expert academic computational biologist with decades of
+    role = "Computational Biologist and Geneticist"
+    goal = (
+        "Analyze and interpret datasets, and make suggestions into next analysis steps."
+    )
+    backstory = """You are an expert academic computational biologist with decades of
     experience working in microbial genetics. You have published several genome announcement
     papers and have worked extensively with novel sequence data."""
+    _openai_key: str
+    _catalog_db_dir: Path
+    _docs_db_dir: Path
 
-    def __init__(self: "AnalystAgent", token: str, llm: LLM):
+    def __init__(
+        self: "AnalystAgent",
+        token: str,
+        llm: LLM,
+        openai_api_key: str = None,
+        catalog_db_dir: Path | str = None,
+        docs_db_dir: Path | str = None,
```

Comment:
These are now parameterized as kwargs. Should make it easier for testing with different options, though the defaults still remain.